File: 0740-delete-and-earn/0740-delete-and-earn.py
class Solution:
    def deleteAndEarn(self, nums: List[int]) -> int:
        _max = max(nums)
        freq, dp = [0] * (_max + 1), [0] * (_max + 1)
        for n in nums:
            freq[n] += 1
        for i in range(1, _max+1):
            dp[i] = max(dp[i-1], dp[i-2] + i * freq[i])
        return dp[_max]

File: 0740-delete-and-earn/0740-delete-and-earn.ts
function deleteAndEarn(nums: number[]): number {
    const max = Math.max(...nums)
    const dp: number[] = Array(max+1).fill(0)
    const freq: number[] = Array(max+1).fill(0)
    for (const n of nums) {
        freq[n]++
    }
    dp[1] = 1 * freq[1]
    for (let i=2; i <= max; i++) {
        dp[i] = Math.max(dp[i-1], dp[i-2] + i * freq[i])
    }
    return dp[max]
};